src/cons/cons.cpp

```diff
@@ -425,6 +535,10 @@ void timewait_stage(bool reset)
 void apply_ledger(const p2p::proposal &cons_prop)
 {
     // todo:write lcl.
+    ctx.led_seq_no++;
+    std::string lcl = cons::save_ledger(cons_prop, ctx.led_seq_no);
```

Comment:
There are 2 copy operations happening here. In line 539 and 541. Just assign the `save_ledger()` return value to `ctx.lcl`.

Reply:
yes. That was a stupid mistake.